src/Ocelot.Provider.Polly/PollyQoSProvider.cs

```diff
@@ -39,25 +29,30 @@ public PollyQoSProvider(DownstreamRoute route, IOcelotLoggerFactory loggerFactor
                         durationOfBreak: TimeSpan.FromMilliseconds(route.QosOptions.DurationOfBreak),
                         onBreak: (ex, breakDelay) =>
                         {
-                            _logger.LogError(
+                            logger.LogError(
                                 ".Breaker logging: Breaking the circuit for " + breakDelay.TotalMilliseconds + "ms!", ex);
                         },
                         onReset: () =>
                         {
-                            _logger.LogDebug(".Breaker logging: Call ok! Closed the circuit again.");
+                            logger.LogDebug(".Breaker logging: Call ok! Closed the circuit again.");
                         },
                         onHalfOpen: () =>
                         {
-                            _logger.LogDebug(".Breaker logging: Half-open; next call is a trial.");
+                            logger.LogDebug(".Breaker logging: Half-open; next call is a trial.");
                         }
                     );
             }
             else
             {
-                _circuitBreakerPolicy = null;
+                circuitBreakerPolicy = null;
             }
 
-            CircuitBreaker = new CircuitBreaker(_circuitBreakerPolicy, _timeoutPolicy);
+            CircuitBreaker = new CircuitBreaker(circuitBreakerPolicy, timeoutPolicy);
+        }
+
+        [Obsolete("Use the constructor PollyQoSProvider(DownstreamRoute route, IOcelotLoggerFactory loggerFactory)")]
+        public PollyQoSProvider(AsyncCircuitBreakerPolicy circuitBreakerPolicy, AsyncTimeoutPolicy timeoutPolicy, IOcelotLogger logger)
+        {

```

Comment:
```suggestion
        private const string ObsoleteConstructorMessage = $"Use the constructor {nameof(PollyQoSProvider)}({nameof(DownstreamRoute)} route, {nameof(IOcelotLoggerFactory)} loggerFactory)!";

        [Obsolete(ObsoleteConstructorMessage)]
        public PollyQoSProvider(AsyncCircuitBreakerPolicy circuitBreakerPolicy, AsyncTimeoutPolicy timeoutPolicy, IOcelotLogger logger)
        {
            throw new NotSupportedException(ObsoleteConstructorMessage);
```
And it is strange that obsolete constructor has 0 (zero) references!...
Were there zero references?
If Not, all referenced scopes should be fixed, and addition to DI. But probably the constructor is not referenced at all.